src/common/factory/emailDeliveryIntegrator/emailDeliveryIntegrator.factory.ts

```diff
@@ -0,0 +1,35 @@
+import { SendgridEmailDeliveryIntegrator } from './sendgridEmailDeliveryIntegrator';
+import { EmailType } from '../../../email/core/email-type';
+import {
+  EmailKeyHost,
+  EmailRegistration,
+} from '../../../email/core/register-email';
+import { SenderOptions } from 'src/email/core/sender-options';
+
+export interface IEmailDeliveryIntegrator {
```

Comment:
I'd recommend implementing this as an abstract class. The advantage is that abstract classes can have method implementations by default, meaning that you can provide some behavior out of the box. It may not amount to much in this case, but it's a good practice in general.